src/www/ui/api/Controllers/UploadController.php

```diff
@@ -345,7 +350,14 @@ private function changeUpload($request, $response, $args, $isCopy)
    */
   public function postUpload($request, $response, $args)
   {
+
+    $reqBody = $this->getParsedBody($request);
+    if (empty($reqBody["scanOptions"])) {
+      $error = new Info(403, "No agents selected!", InfoType::ERROR);
```

Comment:
It is not required for the user to select any agent during the upload. This will be contradicting with the current UI.

Reply:
Yes, this's because I want to create the corresponding job immediately after the upload-creation on the backend. So i want to receive all the required arguments onwards including those for the job I will do later; so as to execute the two processes simultaneously (one after the other).

So in order to continue with the job process, I'm required to specify  those  scanOptions eventhough it's properties might all be set to false.

In order words; I've followed the same flow for creating the job for the upload referring to this job controller's function.

![Menu_002](https://user-images.githubusercontent.com/66276301/183885215-63cdb23a-dc0c-48ae-b8a6-d759e61bb2af.png)

Reply:
let me know if I'm right sir

Reply:
Not exactly. The other endpoint is built specifically to schedule agents on an upload. So it there are no agents selected by the user to schedule, the endpoint will not do any thing. Therefore, it send `403` if no agent is selected.

However, the upload endpoint still have some tasks to do even if user selects no agents to schedule. The same principle is followed in current UI. You can select no agent in section 7 (aptly names "Select **optional** analysis)
![image](https://user-images.githubusercontent.com/18077542/184351397-b08ce782-a9f9-472e-8faf-354000b727a4.png)

Where as while scheduling agents on an existing upload, the section is called "Select additional analysis". You can visit the UI for it by visiting [Jobs > Schedule Agents](http://localhost/repo/?mod=agent_add)
![image](https://user-images.githubusercontent.com/18077542/184351481-4ef68b3b-606f-4c0b-b607-5c135cbc5301.png)